src/helpers/NameHelpers.ts

```diff
@@ -0,0 +1,16 @@
+import { CommonModel } from 'models';
+
+/**
+ * Recursively find the proper property name for additionalProperties
+ * 
+ * This function ensures that the property name for additionalProperties is unique 
+ * 
+ * @param rootModel 
+ * @param additionalPropertyName 
+ */
+export function findPropertyNameForAdditionalProperties(rootModel: CommonModel, additionalPropertyName = 'additionalProperties'): string {
```

Comment:
`findPropertyNameForAdditionalProperties` Java in JS 😂 Maybe another name like 

![image](https://user-images.githubusercontent.com/20404945/122389209-fe9e3500-cf70-11eb-9017-5c0cf7823e8c.png)

```suggestion
export function getUniquePropertyName(rootModel: CommonModel, propertyName = 'additionalProperties'): string {
```

Reply:
hahahaha, true 😂